src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -64,284 +59,33 @@ std::vector<char> xma_xclbin_file_open(const std::string& xclbin_name)
-    if (xclbin_info->number_of_mem_banks > MAX_DDR_MAP) {
-        xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "XMA supports max of only %d mem banks ", MAX_DDR_MAP);
-        throw std::runtime_error("XMA supports max of only " + std::to_string(MAX_DDR_MAP) + " mem banks");
-    }
-    for (int i = 0; i < mem_topo->m_count; i++) {
-        XmaMemTopology temp_mem_topology;
-        temp_mem_topology.m_type = mem_topo->m_mem_data[i].m_type;
-        temp_mem_topology.m_used = mem_topo->m_mem_data[i].m_used;
-        temp_mem_topology.m_size = mem_topo->m_mem_data[i].m_size;
-        temp_mem_topology.m_base_address = mem_topo->m_mem_data[i].m_base_address;
-        //m_tag is 16 chars
-        auto tmp_ptr = mem_topo->m_mem_data[i].m_tag;
-        temp_mem_topology.m_tag = {tmp_ptr, std::find(tmp_ptr, tmp_ptr+16, '\0')};//magic 16 from xclbin.h
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "index=%d, tag=%s, type = %ud, used = %ud, size = %lx, base = %lx ",
-               i,temp_mem_topology.m_tag.c_str(), temp_mem_topology.m_type, temp_mem_topology.m_used,
-               temp_mem_topology.m_size, temp_mem_topology.m_base_address);
-        xma_mem_topology.emplace_back(std::move(temp_mem_topology));
-    }
-
-    return XMA_SUCCESS;
-}
-
-static int get_xclbin_connectivity(const char *buffer, XmaXclbinInfo *xclbin_info)
-{
-    const axlf *xclbin = reinterpret_cast<const axlf *>(buffer);
-
-    const axlf_section_header *ip_hdr = xrt_core::xclbin::get_axlf_section(xclbin, ASK_GROUP_CONNECTIVITY);
-    if (ip_hdr)
-    {
-        const char *data = &buffer[ip_hdr->m_sectionOffset];
-        const connectivity *axlf_conn = reinterpret_cast<const connectivity *>(data);
-        auto& xma_connectivity = xclbin_info->connectivity;
-        xclbin_info->number_of_connections = axlf_conn->m_count;
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "CONNECTIVITY - %d connections ",xclbin_info->number_of_connections);
-        for (int i = 0; i < axlf_conn->m_count; i++)
-        {
-            XmaAXLFConnectivity temp_conn;
-
-            temp_conn.arg_index         = axlf_conn->m_connection[i].arg_index;
-            temp_conn.m_ip_layout_index = axlf_conn->m_connection[i].m_ip_layout_index;
-            temp_conn.mem_data_index    = axlf_conn->m_connection[i].mem_data_index;
-            xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "index = %d, arg_idx = %d, ip_idx = %d, mem_idx = %d ",
-                     i, temp_conn.arg_index, temp_conn.m_ip_layout_index,
-                     temp_conn.mem_data_index);
-            xma_connectivity.emplace_back(std::move(temp_conn));
-        }
-    }
-    else
-    {
-        xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not find CONNECTIVITY in xclbin ip_hdr=%p ", ip_hdr);
+    const axlf_section_header* conn_hdr = xrt_core::xclbin::get_axlf_section(xclbin_ax, ASK_GROUP_CONNECTIVITY);
+    if (!conn_hdr) {
+        xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not find CONNECTIVITY in xclbin conn_hdr=%p ", conn_hdr);
```

Comment:
axlf parsing?

Reply:
addressed this comment and used XRT::XCLBIN class in the code.